### PR TITLE
Adjust USB vs. serial telemetry precedence in flight code

### LINKED ISSUE #630: Serial telemetry does not work on LUX
Suspect it's related to USB presence detect or lack thereof...

### PR DESCRIPTION
**Pull request by mlyle**
If we've received USB bytes recently, then be willing to use USB for communications.  Otherwise disqualify USB.

fixes #630 #325

### CODE REVIEW

**Comment by mlyle** [comment]
Have not gotten to test this; just wrote the code.

**Comment by mlyle** [comment]
Tested on Lux.  Tested on revolution.  (Was able to switch back and forth between USB and RFM22 when powered by USB).  Tested on coptercontrol-- able to switch back and forth between USB and bluetooth when powered by USB.

File: flight/PiOS/Common/pios_com.c [hunk 1]
```diff
@@ -463,6 +463,36 @@ int32_t PIOS_COM_SendFormattedString(uintptr_t com_id, const char *format, ...)
 }
 
 /**
+ * Reports number of bytes available for receiving.
+ * \param[in] com_id the COM instance to receive from
+ * \returns number of bytes available to be read
+ */
+uint16_t PIOS_COM_GetNumReceiveBytesPending(uintptr_t com_id) {
+	struct pios_com_dev * com_dev = (struct pios_com_dev *)com_id;
```
Comment (tracernz) [inline flight/PiOS/Common/pios_com.c]: *com_dev

**Comment by tracernz** [comment]
Thanks, this will be great!

File: flight/PiOS/Common/pios_com.c (hunk 1 above)
Comment (mlyle) [inline flight/PiOS/Common/pios_com.c]: Man, I always get in trouble for the lines i copy and paste :P

**Comment by mlyle** [comment]
Did a global search and replace for 'pios_com_dev \* '

**Comment by tracernz** [comment]
> Did a global search and replace for 'pios_com_dev \* '

Nice. :)

**Comment by mlyle** [comment]
@tracernz -- want to clarify the 'requires test'?  Do you want me to test again after the formatting changes?  Do you want third-party test?

**Comment by tracernz** [comment]
Third party ideally. I'm not able so if nobody else does soon I'm okay with it merging anyway.

**Comment by mlyle** [comment]
There was indeed a defect.  Corrected and flight tested on Sparky2 as part of kitchensink branch.  I think this is ready for merge.

**Comment by mlyle** [comment]
Was OK on Naze32.

**Pull request merged**